lib/rules/media-type-no-deprecated/index.mjs

```diff
@@ -0,0 +1,73 @@
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import { atRuleParamIndex } from '../../utils/nodeFieldIndices.mjs';
+import { atRuleRegexes } from '../../utils/regexes.mjs';
+import { deprecatedMediaTypesNames } from '../../reference/mediaTypes.mjs';
+import { isMediaQueryWithType } from '@csstools/media-query-list-parser';
+import optionsMatches from '../../utils/optionsMatches.mjs';
+import parseMediaQuery from '../../utils/parseMediaQuery.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'media-type-no-deprecated';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (mediaType) => `Unexpected deprecated media type "${mediaType}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/media-type-no-deprecated',
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				actual: secondaryOptions,
+				possible: {
+					ignoreMediaType: [isString, isRegExp],
+				},
+				optional: true,
+			},
+		);
+
+		if (!validOptions) return;
+
+		root.walkAtRules(atRuleRegexes.mediaName, (atRule) => {
+			const mediaQueryList = parseMediaQuery(atRule);
+
+			mediaQueryList.forEach((mediaQuery) => {
+				if (!isMediaQueryWithType(mediaQuery)) return;
+
+				const mediaType = mediaQuery.getMediaType();
+
+				if (!deprecatedMediaTypesNames.has(mediaType.toLowerCase())) return;
+
+				if (optionsMatches(secondaryOptions, 'ignoreMediaType', mediaType)) return;
+
+				const index = atRuleParamIndex(atRule);
+				const endIndex = index + mediaType.length;
```

Comment:
major

The way this is calculated is wrong.
i.e. it assumes it's always the first one
try `@media screen, print, tty {}` in [the demo](https://stylelint.io/demo/#N4IgJgpgDgziBcJgB0B2ACdyQwC4E8AbCQgS1V23ixAAtddZ4B6ZqAawHMA6KAJ26oIAd2Z4iJcrgACAFgAcsiPICGARmwAaNJmzjiZCgFoAxgHtUAM1KcjeFajAq+YKjUIrcEPNjQBfED8gA)

Reply:
see https://github.com/stylelint/stylelint/pull/8668/files#r2203390243